compiler/rustc_borrowck/src/diagnostics/region_name.rs

```diff
@@ -840,4 +851,41 @@ impl<'tcx> MirBorrowckCtxt<'_, 'tcx> {
             source: RegionNameSource::AnonRegionFromYieldTy(yield_span, type_name),
         })
     }
+
+    fn give_name_if_anonymous_region_appears_in_impl_signature(
+        &self,
+        fr: RegionVid,
+    ) -> Option<RegionName> {
+        let ty::ReEarlyBound(region) = *self.to_error_region(fr)? else {
+            return None;
+        };
+        if region.has_name() {
+            return None;
+        };
+
+        let tcx = self.infcx.tcx;
+        let impl_did = tcx.impl_parent(self.mir_def_id())?;
+        if tcx.parent(region.def_id) != impl_did {
+            return None;
+        }
+
+        let mut found = false;
+        tcx.fold_regions(tcx.type_of(impl_did), &mut true, |r: ty::Region<'tcx>, _| {
+            if *r == ty::ReEarlyBound(region) {
+                found = true;
+            }
+            r
+        });
+
+        Some(RegionName {
+            name: self.synthesize_region_name(),
+            source: RegionNameSource::AnonRegionFromImplSignature(
+                tcx.def_span(region.def_id),
+                // FIXME(compiler-errors): Does this ever actually show up
+                // anywhere other than the self type? I couldn't create an
+                // example of a `'_` in the impl's trait being referenceable.
```

Comment:
It can appear in the trait:
```rust
impl Foo<'_> for () {}
```

Reply:
I know that elided lifetimes show up in that position, but I had trouble recreating this ICE using an elided lifetime in the trait generics. 

In order to use that lifetime to produce an error, it needs to show up somewhere as an input or output in the method signature as well, which means we are able to give the lifetime a name and never hit this function.

Reply:
This implementation would give a name if it's ever needed.  Let's keep it as such, and wait for somebody to ask for a better message with an example.